### PR TITLE
PhpdocSeparationFixer - Ignore incorrect PHPDoc

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2365

@GrahamCampbell happy debugging!

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Ah, thank you. :)

**Comment by GrahamCampbell** [comment]
Well, I mean, naively just reverting to the original algorithm if the new one doesn't match the correct number of items works:

image

image

**Comment by SpacePossum** [comment]
:+1:  however it would be nice if we could avoid the second `preg_match*` though

File: tests/Fixer/Phpdoc/PhpdocSeparationFixerTest.php [hunk 1]
```diff
@@ -24,6 +24,11 @@
 {
     public function testFix()
     {
+        $this->doTest('<?php
+/** @param EngineInterface $templating
```
Comment (keradus) [inline tests/Fixer/Phpdoc/PhpdocSeparationFixerTest.php]: i would not treat it as proper docblock

still, tool must not crash on it, but i woulnd't say it's crucial to "fix" it

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Comment by GrahamCampbell** [comment]
I don't like the way this fix is implemented. You made the line class silently fail. It should be the responsibility of the fixer to decide what to do imo.

**Comment by keradus** [comment]
we quickly fix the tool to not crash, which was the main concern.
if you want to provide a fix to change phpdoc in that scenario - go ahead ;)

**Pull request merged**